pydy/tests/test_joints.py

```diff
@@ -0,0 +1,275 @@
+        assert self.pinjoint.child_joint_vector == child_point
+
+        # part 3 aligning parent_axis and child_axis
+        self.join_frames()
+        # sample implementation
+        # self.child.frame.orient(self.parent.frame, 'Axis', [0, self.parent.frame.x])
+        assert cross(self.pinjoint.parent_axis, self.pinjoint.child_axis) == self.child.frame.z
+        assert cross(self.pinjoint.child_axis, self.pinjoint.parent_axis) == - self.parent.frame.z
+        self.align_axes()
+        parent_axis = self.pinjoint.parent_axis
+        child_axis = self.pinjoint.child_axis
+        # sample implementation
+        # perpendicular_axis_in_parent = - cross(parent_axis, child_axis)
+        # angle_between_axes = acos(dot(self.pinjoint.child_axis, self.pinjoint.parent_axis))
+        # self.child.frame.orient(self.parent.frame, 'Axis',
+        #                        [acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())),
+        #                         cross(parent_axis, child_axis)])
+        assert acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())) == 0
+
+        # part 4 adding angular velocity to child w.r.t parent.
+        assert self.child.frame.ang_vel_in(self.parent.frame) != 0
+
+    def test_pinjoint_apply_joint(self):
+        # apply_joint() should do everything done above by calling specific funtions
+        self.new_pinjoint = PinJoint('pinjoint', self.parent, self.child)
+        self.new_pinjoint._apply_joint()
+        # part 1
+        assert self.child.parent == self.parent
+        assert self.parent.child == self.child
+
+        # part 2
+        child_masscenter = self.child.masscenter
+        parent_masscenter = self.parent.masscenter
+        parent_point = parent_masscenter.locatenew(self.name + '_Point', self.parent_joint_vector)
+        child_point = child_masscenter.locatenew(self.name + '_Point', self.child_joint_vector)
+        assert self.pinjoint.parent_joint_vector == parent_point
+        assert self.pinjoint.child_joint_vector == child_point
+
+        # part 4
+        parent_axis = self.pinjoint.parent_axis
+        child_axis = self.pinjoint.child_axis
+        assert acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())) == 0
+
+
+class TestSlidingJoint():
+    def setup(self):
+        self.parent = Body('parent')
+        self.child = Body('child')
+        self.slidingjoint = SlidingJoint('slidingjoint', self.parent, self.child)
+
+    def test_slidingjoint_init(self):
+        # Default values
+        assert self.slidingjoint.parent == self.parent
+        assert self.slidingjoint.child == self.child
+        assert self.slidingjoint.direction1 == self.parent.x
+        assert self.slidingjoint.direction2 == self.child.x
+        assert self.slidingjoint.parent_point == self.parent.masscenter
+        assert self.slidingjoint.child_point == self.child.masscenter
+
```

Comment:
If these attributes are public can we change them after `__ini__`? This is possible in the system class, for example. It is a nice feature to have.